scripts/common.py

```diff
@@ -712,3 +715,99 @@ def __enter__(self):
 
     def __exit__(self, etype, value, traceback):
         os.chdir(self.saved_path)
+
+
+@contextlib.contextmanager
+def managed_process(command_args, shell=False, timeout_secs=60, **kwargs):
+    """Context manager for starting and stopping a process gracefully.
+
+    Args:
+        command_args: list(int | str). A sequence of program arguments, where
+            the program to execute is the first item.
+        shell: bool. Whether the command should be run inside of its own shell.
+        timeout_secs: int. The time allotted for the managed process and its
+            descendants to terminate themselves. After the timeout, any
+            remaining processes will be killed abruptly.
+        **kwargs: dict(str: *). Same kwargs as `subprocess.Popen`.
+
+    Yields:
+        psutil.Process. The process managed by the context manager.
+    """
+    if PSUTIL_DIR not in sys.path:
+        sys.path.insert(1, PSUTIL_DIR)
+    import psutil
+
+    # We're only using these generators to reduce code duplication.
```

Comment:
I don't understand this comment... what alternative are you envisioning?

Reply:
```python
command = (
    ' '.join(str(a).strip() for a in stripped_args if str(a).strip()))
    if shell else
    list(str(a).strip() for a in stripped_args if str(a).strip()))
```

Reply:
Ah ok. I suggest just dropping that comment, since without that context it's actually kind of unclear what you're referring to and I think it's not worth keeping it. (In general, we try to write most code in a way that reduces code duplication.)

Reply:
Ack'd, will resolve in follow-up PR.